### PR TITLE
Bots

### LINKED ISSUE #21: Migrate Zoom plugin to use bot accounts
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Plugin: Jira" community channel](https://community.mattermost.com/core/channels/jira-plugin).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/) and our [Plugins Guide](https://developers.mattermost.com/extend/plugins/).

----

Migrate Zoom plugin to use bot accounts.

For an example, here is a PR that migrated the Demo Plugin to use bot accounts with a helper function https://github.com/mattermost/mattermost-plugin-demo/pull/29

### PR DESCRIPTION
**Pull request by hanzei**
This PR:
- Fixes the modules name
- Migrates to bot accounts
- Refactors some code in `server`
- Adds additional error handling

The best way to review the code is my reviewing the individual commits.

If the refactoring changes are appreciated, I can also apply them to `server/zoom`.

Fixes #21

### CODE REVIEW

**Comment by hanzei** [comment]
The quality of `assets/profile.png` is pretty bad. Is there a different file I can use?

**Comment by jasonblais** [comment]
> The quality of assets/profile.png is pretty bad. Is there a different file I can use?

Tagging @asaadmahmood for this question

**Comment by asaadmahmood** [comment]
@hanzei Here's one with a larger resolution.
Zoom

File: server/plugin_test.go [hunk 1]
```diff
@@ -31,15 +32,20 @@ func TestPlugin(t *testing.T) {
 
 			str, _ := json.Marshal(user)
 
-			w.Write([]byte(str))
+			if _, err := w.Write(str); err != nil {
```
Comment (levb) [inline server/plugin_test.go]: nit: save the `if`, require will panic if err anyway

Comment (hanzei) [reply server/plugin_test.go]: How do you know this?

**Pull request merged**